lib/logstash/outputs/s3.rb

```diff
@@ -135,6 +135,9 @@ class LogStash::Outputs::S3 < LogStash::Outputs::Base
  # 0 stay all time on listerner, beware if you specific 0 and size_file 0, because you will not put the file on bucket,
  # for now the only thing this plugin can do is to put the file when logstash restart.
  config :time_file, :validate => :number, :default => 0 
+
+ #Set the directory where temporary files are stored.
+ config :temp_directory, :validate => :string, :default => "/opt/logstash/S3_temp/"
```

Comment:
I think the default should be the system's temporary directory. `Dir.tmpdir` usually reports this.